packages/esbuild/src/executors/esbuild/esbuild.impl.ts

```diff
@@ -68,7 +72,7 @@ export async function* esbuildExecutor(
     ...options,
     // TODO(jack): make types generate with esbuild
     skipTypings: true,
-    outputFileExtensionForCjs: CJS_FILE_EXTENSION,
+    outputFileExtensionForCjs: getOutExtension('cjs', options),
```

Comment:
This is the main fix. Use the same function used in other places to determine the CJS file extension.